d2/src/main/java/com/linkedin/d2/balancer/simple/SimpleLoadBalancer.java

```diff
@@ -931,7 +929,7 @@ public void getLoadBalancedClusterAndUriProperties(String clusterName, boolean w
     }
   }
 
-  private void handleTimeoutFromGetClusterAndUriProperties(Throwable e, String clusterName,
+  public void handleTimeoutFromGetClusterAndUriProperties(String clusterName,
```

Comment:
Does this only need to have default visibility for tests? Can we also add the `@VisibleForTesting` annotation?

Reply:
Yes. Added